### ISSUE TITLE
Percona-XtraDB-Cluster prohibits use of GET_LOCK with pxc_strict_mode = ENFORCING

### ISSUE THREAD

**Issue by danieltwork** [prisma/prisma#16020]
### Bug description

Trying to perform a migration on a production database.

It reporting an error:

```
node@worker:/$ node_modules/prisma/build/index.js migrate deploy
Prisma schema loaded from prisma/schema.prisma
Datasource "db": MySQL database "test-database" at "onprem-xtradb-cluster-haproxy.my-organization.svc.cluster.local:3306"

5 migrations found in prisma/migrations

Error: Percona-XtraDB-Cluster prohibits use of GET_LOCK with pxc_strict_mode = ENFORCING
   0: migration_core::state::ApplyMigrations
             at migration-engine/core/src/state.rs:185
```

For example, TypeORM does not report such an error.
What am I doing wrong?

### How to reproduce

Start Prisma migration to a Percona Server for MySQL database.

### Prisma information

```prisma
generator client {
  provider = "prisma-client-js"
  binaryTargets = ["native", "linux-arm64-openssl-1.1.x", "linux-musl"]
}

datasource db {
  provider = "mysql"
  url      = env("STORAGE_MYSQL_DSN")
}

model Token {
  @@map("token")
  id    Int     @default(autoincrement()) @id
  user_id String @unique
  token Json?
}
```

### Environment & setup

- OS: Kubernetes
- Database: Percona Server for MySQL database

### Prisma Version

```
prisma & client: 4.2.1
```

**Comment by Jolg42** [comment]
Some context about this error from Percona
https://forums.percona.com/t/we-have-installed-magento-2-3-4-with-parcona-xtradb-cluster-and-we-are-getting-error-on-cron-run/7881/2
>The error message you are receiving is quite self-explanatory. The function GET_LOCK() is not allowed in PXC while the cluster is in ENFORCING mode. Your options are to change pxc_strict_mode=PERMISSIVE, or modify the behavior of the application to stop using GET_LOCK() functions and use more app-native locking.
[https://www.percona.com/doc/percona-xtradb-cluster/LATEST/features/pxc-strict-mode.html 41](https://www.percona.com/doc/percona-xtradb-cluster/LATEST/features/pxc-strict-mode.html)
The reason this function is prohibited is because this function is not cluster-wide. If your application executes GET_LOCK() on node1, another connection on node2() can acquire the same lock without conflict. If you are certain that your application is isolated to 1 node, then you should feel safe in turning off this protection.

Docs
https://docs.percona.com/percona-xtradb-cluster/8.0/features/pxc-strict-mode.html

I think the origin is that we perform "Advisory Locking" in Migrate to avoid concurrent executions of Migrate commands on the same database, see https://www.prisma.io/docs/concepts/components/prisma-migrate/migrate-development-production#advisory-locking

It looks like the potential solution could be:
- change the `pxc_strict_mode` setting
- or make it possible to configure Migrate to skip the "Advisory Locking"

**Comment by tatupesonen** [comment]
Any further work on this?

**Comment by MateusRoscoe** [comment]
Any word on this?

**Comment by Jolg42** [comment]
It has been a bit quiet here for sure. No news at the moment, our process is that we would need to do a reproduction first.
My guess is that it should be reproducible with the information provided.

After that, we would need to think about how to handle this case and implement it, which I think means being able to disable the `Advisory Locking` somehow?

If someone is motivated to work on this, I think it might be a relatively easy thing to dig into
The code for the locking is in https://github.com/prisma/prisma-engines/blob/1548dc15e415a2298c25af9b04f33bba3d28b998/schema-engine/connectors/sql-schema-connector/src/flavour/mysql.rs#L71-L88

Changing the `pxc_strict_mode` setting is not an option, I guess?

**Comment by tatupesonen** [comment]
Our solution has been to change `pxc_strict_mode` to `PERMISSIVE` but it may have some adverse effects if running in clustering mode.

**Comment by Jolg42** [comment]
Note:
I could reproduce with
```
docker run -d -p 3306:3306 -e MYSQL_ROOT_PASSWORD=mypass -e CLUSTER_NAME=something -e XTRABACKUP_PASSWORD=mypass2 percona/percona-xtradb-cluster
```

and using this as the connection string
```
DATABASE_URL=mysql://root:mypass@127.0.0.1:3306/mydb6
```

I verified that `SET GLOBAL pxc_strict_mode=PERMISSIVE;` unblocks the situation like mentioned before.

**Comment by Jolg42** [comment]
In https://github.com/prisma/prisma-engines/pull/4171 I added a new environment variable to disable the advisory locking.

Example values that disable the locking 
```
PRISMA_SCHEMA_DISABLE_ADVISORY_LOCK=1
PRISMA_SCHEMA_DISABLE_ADVISORY_LOCK=anything
```

Values that DO NOT disable the locking
```
PRISMA_SCHEMA_DISABLE_ADVISORY_LOCK=
PRISMA_SCHEMA_DISABLE_ADVISORY_LOCK=0
```

Note: This is now fixed in `5.3.0-dev.13` if you want to try it out. Let us know if it works like expected.
It will be released in the official 5.3.0 version on Tuesday, September, 12th.
